experimental/traffic-portal/src/app/shared/import-json-txt/import-json-txt.component.ts

```diff
@@ -0,0 +1,159 @@
+	templateUrl: "./import-json-txt.component.html",
+})
+export class ImportJsonTxtComponent {
+
+	/**
+	 * Allowed import file types
+	 */
+	public allowedType: string[] = ["application/json", "text/plain"];
+
+	/** Text editor value */
+	public inputTxt = "";
+
+	/**  File data imported */
+	public fileData = "";
+
+	/** Monitor whether any file is being drag over the dialog */
+	public dragOn = false;
+
+	/**
+	 * Creates an instance of import json edit txt component.
+	 *
+	 * @param dialogRef Dialog manager
+	 * @param alertService Alert service manager
+	 * @param datePipe Default angular date pipe for formating date
+	 */
+	constructor(
+		@Inject(MAT_DIALOG_DATA) public readonly data: ImportJsonTxtComponentModel,
+		private readonly dialogRef: MatDialogRef<ImportJsonTxtComponent>,
+		private readonly alertService: AlertService,
+		private readonly datePipe: DatePipe) { }
+
+	/**
+	 * Emits the json value for import as profile data
+	 */
+	public submit(): void {
+		this.dialogRef.close(this.inputTxt);
+	}
+
+	/**
+	 * Hosts listener for drag over
+	 *
+	 * @param evt Drag events data
+	 */
+	@HostListener("dragover", ["$event"]) public onDragOver(evt: DragEvent): void {
+		evt.preventDefault();
+		evt.stopPropagation();
+
+		this.dragOn = true;
+	}
+
+	/**
+	 * Hosts listener for drag leave
+	 *
+	 * @param evt Drag events data
+	 */
+	@HostListener("dragleave", ["$event"]) public onDragLeave(evt: DragEvent): void {
+		evt.preventDefault();
+		evt.stopPropagation();
+
+		this.dragOn = false;
+	}
+
+	/**
+	 * Hosts listener for drop
+	 *
+	 * @param evt Drag events data
+	 */
+	@HostListener("drop", ["$event"]) public onDrop(evt: DragEvent): void {
+		evt.preventDefault();
+		evt.stopPropagation();
+
+		this.dragOn = false;
+
+		const file = evt.dataTransfer?.files[0];
+
+		// returns on when there is no file attachment is there
+		if (!file) {
+			return;
+		}
+
+		this.docReader(file);
+	}
+
+	/**
+	 * Uploads file
+	 *
+	 * @param event Event object for upload file
+	 */
+	public uploadFile(event: Event): void {
+		const file = (event.target as HTMLInputElement).files?.[0];
+
+		// returns on when there is no file attachment is there
+		if (!file) {
+			return;
+		}
+		this.docReader(file);
+	  }
+
+	/**
+	 * Docs reader
+	 *
+	 * @param file that is uploaded
+	 */
+	public docReader(file: File): void {
+
+		/**
+		 * Check whether expected file is being uploaded
+		 * returns on file wrong file type is uploaded
+		 */
+		if (!this.allowedType.find(type => type === file.type)) {
+			this.alertService.newAlert({ level: AlertLevel.ERROR, text: "Only JSON or text file is allowed." });
+			return;
+		}
+
+		/** Format text with data from file data and formated date with date pipe */
+		this.fileData = `${file.name} - ${file.size} bytes, last modified: ${this.datePipe.transform(file.lastModified, "MM-dd-yyyy")}`;
+
+		const reader = new FileReader();
+
+		reader.onload = (event): void => {
+			this.inputTxt = event.target?.result as string;
```

Comment:
please add a comment explaining why you need to use `as` here.

Reply:
add a condition now brennan.